controllers/osbuildenvconfig_controller_test.go

```diff
@@ -893,26 +899,81 @@ var _ = Describe("OSBuildEnvConfig Controller", func() {
 													})
 												})
 
-												Context("ConfigMap for the osbuild-worker config exists", func() {
+												Context("ConfigMap for the ansible config exists", func() {
```

Comment:
I know that we already talked about that - but it become difficult to track the test cases with that adaptation

Reply:
Not sure what you're referring to.
The root cause of this complicated unit test is the fact that we have to test the main function (Reconcile) as a whole. We cannot test the internal functions since the test is in a different package, and we can't mock the internal methods to contain the tests of the external one.
Furthermore, with all the contexts I defined, when a test fails, it's really easy to understand where in the flow you are.